Content.Shared/Clothing/Components/ClothingComponent.cs

```diff
@@ -60,9 +67,15 @@ public sealed partial class ClothingComponent : Component
     public string? RsiPath;
 
     /// <summary>
-    /// Name of the inventory slot the clothing is in.
+    /// Name and flags of the inventory slot the clothing is currently in.
+    /// Note that this being non-null does not mean the clothing is considered "worn" or "equipped" unless the slot
+    /// satisfies the <see cref="Slots"/> flags.
     /// </summary>
-    public string? InSlot;
+    [DataField]
+    public (string Name, SlotFlags Flag)? InSlot;
```

Comment:
This is a cleaner solution, I agree. But I'd wanted to avoid changing `InSlot` itself purely because I wanted to ensure that if it's being used somewhere downstream this won't break anything.